src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -1622,7 +1585,7 @@ void EmitWhenHasCapture()
                         writer.WriteLine($"matchIndex = base.MatchIndex({capnum});");
                         using (EmitBlock(writer, $"for (int i = 0; i < matchLength; i++)"))
                         {
-                            using (EmitBlock(writer, $"if ({ToLowerIfNeeded(hasTextInfo, options, $"inputSpan[matchIndex + i]", caseInsensitive)} != {ToLowerIfNeeded(hasTextInfo, options, $"inputSpan[pos - matchLength + i]", caseInsensitive)})"))
+                            using (EmitBlock(writer, $"if ({ToLowerIfNeeded($"inputSpan[matchIndex + i]", caseInsensitive)} != {ToLowerIfNeeded($"inputSpan[pos - matchLength + i]", caseInsensitive)})"))
```

Comment:
Similarly, we shouldn't need the ToLowerIfNeeded helper anymore, if this is the only place it's used.  This can be:
```suggestion
                            using (EmitBlock(writer, caseInsensitive ?
                                $"if (_textInfo.ToLower(inputSpan[matchIndex + i]) != _textInfo.ToLower(inputSpan[pos - matchLength + i]))" :
                                $"if (inputSpan[matchIndex + i] != inputSpan[pos - matchLength + i])")
```